src/test/rustdoc-json/enum_variant_hidden.rs

```diff
@@ -0,0 +1,9 @@
+#![no_std]
+
+// @has enum_variant_hidden.json "$.index[*][?(@.name=='ParseError')]"
+// @has enum_variant_hidden.json "$.index[*][?(@.name=='UnexpectedEndTag')]"
+// @!has enum_variant_hidden.json "$.index[*][?(@.name=='u32')]"
```

Comment:
This doesn't test that the field was hidden, as

```rust
// @has enum_variant_hidden.json "$.index[*][?(@.name=='ParseError')]"
// @has enum_variant_hidden.json "$.index[*][?(@.name=='UnexpectedEndTag')]"
// @!has enum_variant_hidden.json "$.index[*][?(@.name=='u32')]"

pub enum ParseError {
    UnexpectedEndTag(u32),
}
```
will also pass the test.

Primitives arn't stored in the index, but in a varient of `Type`, so the output of the above is
```json
    "inner": {
        "variant_inner": [
          {"inner": "u32", "kind": "primitive"}
        ],
        "variant_kind": "tuple"
      },
```

Instead, just directly check that the variant doesn't list having any fields.

```suggestion
// @is - "$.index[*][?(@.name=='UnexpectedEndTag')].inner.variant_kind" '"tuple"'
// @is - "$.index[*][?(@.name=='UnexpectedEndTag')].inner.variant_inner" []
```